group1/comm/static/comm_tool.js

```diff
@@ -825,3 +823,230 @@ function downloadFileCallback(data) {
 function uploadFileCallback(data) {
   isUploadFiles = false;
 }
+
+//BEGIN VIDEO CHAT CONTROLLER
+var server_host = window.location.hostname;
+var server_port = window.location.port;
+var base_url = 'http://' + server_host + ':3000/';
+var video_socket = io('http://' + server_host + ':3000');
+var DEFAULT_CHANNEL="some-global-channel-name";
+var USE_AUDIO = true;
+var USE_VIDEO = true;
+var MUTE_AUDIO_BY_DEFAULT = false;
+var local_media_stream = null; /* our own microphone / webcam */
+var peers = {};                /* keep track of our peer connections, indexed by peer_id (aka video_socket.io id) */
+var peer_media_elements = {};
+var ICE_SERVERS = [
+    {url:"stun:stun.l.google.com:19302"}
+];
+
+function init() {
+
+    video_socket.on('connect', function() {
+        console.log("Connected to signaling server");
+        setup_local_media(function() {
+            /* once the user has given us access to their
+             * microphone/camcorder, join the channel and start peering up */
+            join_chat_channel(DEFAULT_CHANNEL, {'whatever-you-want-here': 'stuff'});
+        });
+    });
+    video_socket.on('disconnect', function() {
+        console.log("Disconnected from signaling server");
+        /* Tear down all of our peer connections and remove all the
+         * media divs when we disconnect */
+        for (peer_id in peer_media_elements) {
+            peer_media_elements[peer_id].remove();
+        }
+        for (peer_id in peers) {
+            peers[peer_id].close();
+        }
+
+        peers = {};
+        peer_media_elements = {};
+    });
+    function join_chat_channel(channel, userdata) {
+        video_socket.emit('join', {"channel": channel, "userdata": userdata});
+        //alert("join channel");
```

Comment:
Did you want to keep this commented line?

Reply:
nope. resolved.